stats/stats.go

```diff
@@ -203,24 +205,61 @@ func NewHistogram(Offset float64, Divider float64) *Histogram {
 	return h
 }
 
-// Tradeoff memory for speed (though that also kills the cache so...)
-// this creates an array of 100k (max value) entries
-// TODO: consider using an interval search for the last N big buckets
+// Val2Bucket values are kept in two different structure
+// val2Bucket allows you reach between 0 and 1000 in constant time
+// linearTimeVal2Bucket allows you to reach between 1000 and 100000 in linear time
+// linearTimeVal2Bucket approach saves memory usage but loses speed
+// TODO: change linear time to O(log(N)) with binary search for linearTimeVal2Bucket
 func init() {
-	lastV := int32(lastValue)
-	val2Bucket = make([]int, lastV+1)
+	maxArrayValueIndex := 0
+	for i, value := range histogramBucketValues {
+		if value == maxArrayValue {
+			maxArrayValueIndex = i
+			break
+		}
+	}
+	val2Bucket = make([]int, maxArrayValue)
+	linearTineVal2BucketLen := numValues - maxArrayValueIndex
```

Comment:
typo: Tine -> Time but as mentioned above, linearTimeVal2Bucket is not needed

Reply:
I thought you mention about struct that I used.

Reply:
neither the struct nor a new array is needed